File: server/src/main/java/rs/ac/singidunum/novisad/server/model/RealizacijaPredmeta/PolaganjePredmeta.java
package rs.ac.singidunum.novisad.server.model.RealizacijaPredmeta;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import rs.ac.singidunum.novisad.server.model.predmet.Predmet;
import rs.ac.singidunum.novisad.server.model.student.StudentNaGodini;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class PolaganjePredmeta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Integer konacnaOcena;
    @ManyToOne
    private StudentNaGodini student;
    @ManyToOne
    private Predmet predmet;
}

File: client/src/app/model/polaganjePredmeta.ts
import {StudentNaGodini} from "./studentNaGodini";
import {Predmet} from "./predmet";
import {Polaganje} from "./polaganje";

export interface PolaganjePredmeta {
  id: number,
  konacnaOcena: number,
  student?: StudentNaGodini,
  predmet?: Predmet,
  polaganja: Polaganje[]
}
